src/enftun.c

```diff
@@ -228,11 +254,17 @@ enftun_connect(struct enftun_context* ctx)
             goto out;
     }
 
+    if ((rc = enftun_netlink_connect(&ctx->nl, ctx)) < 0)
+        goto out;
+
+
     if ((rc = enftun_tls_connect(&ctx->tls,
                                  ctx->config.fwmark,
                                  ctx->config.remote_hosts,
-                                 ctx->config.remote_port)) < 0)
-        goto out;
+                                 ctx->config.remote_port,
```

Comment:
Modifying the `enftun_tls_connect` signature to support returning the local_addr and tcp_fd feels a bit ugly.

Instead, read them when needed as `ctx->tls.tcp.local_addr` or `ctx->tls.tcp.fd`.  If  you don't like accessing struct members like that, add accessor functions:
```
const struct addrinfo* enftun_tls_local_address(struct enftun_tls* tls);
int enftun_tls_fd(struct enftun_tls* tls);

// OR

void enftun_tls_local_addr(struct enftun_tls* tls, struct localaddr* dest);
void enftun_tls_fd(struct enftun_tls* tls, int* dest);
```

Then modify the `enftun_conn_state` or `enftun_netlink`  init or start function to take those values as arguments.